hbase_storage/src/test/java/com/splicemachine/si/impl/server/SICompactionStateMutateTest.java

```diff
@@ -0,0 +1,251 @@
+        inputCells.addAll(Arrays.asList(
+                SITestUtils.getMockCommitCell(0x100),
+                SITestUtils.getMockTombstoneCell(0x200),
+                SITestUtils.getMockValueCell(0x100)
+        ));
+        transactions.addAll(Arrays.asList(
+                null,
+                TxnTestUtils.getMockActiveTxn(0x200),
+                TxnTestUtils.getMockCommittedTxn(0x100, 0x110)
+        ));
+        cutPurge.mutate(inputCells, transactions, outputCells);
+        assertThat(outputCells, equalTo(inputCells));
+    }
+
+    @Test
+    public void mutateDoNotPurgeBecauseTombstoneRolledBack() throws IOException {
+        inputCells.addAll(Arrays.asList(
+                SITestUtils.getMockCommitCell(0x100),
+                SITestUtils.getMockTombstoneCell(0x200),
+                SITestUtils.getMockValueCell(0x100)
+        ));
+        transactions.addAll(Arrays.asList(
+                null,
+                TxnTestUtils.getMockRolledBackTxn(0x200),
+                TxnTestUtils.getMockCommittedTxn(0x100, 0x110)
+        ));
+        cutPurge.mutate(inputCells, transactions, outputCells);
+        assertThat(outputCells, hasSize(2));
+        assertThat(outputCells, equalTo(Arrays.asList(
+                SITestUtils.getMockCommitCell(0x100),
+                SITestUtils.getMockValueCell(0x100))));
+    }
+
+    @Test
+    public void mutatePartialPurge() throws IOException {
+        inputCells.addAll(Arrays.asList(
+                SITestUtils.getMockCommitCell(0x300),
+                SITestUtils.getMockCommitCell(0x200),
+                SITestUtils.getMockCommitCell(0x100),
+                SITestUtils.getMockAntiTombstoneCell(0x300),
+                SITestUtils.getMockTombstoneCell(0x200),
+                SITestUtils.getMockValueCell(0x300),
+                SITestUtils.getMockValueCell(0x100)
+        ));
+        TxnView transaction1 = TxnTestUtils.getMockCommittedTxn(0x100, 0x110);
+        TxnView transaction2 = TxnTestUtils.getMockCommittedTxn(0x200, 0x210);
+        TxnView transaction3 = TxnTestUtils.getMockCommittedTxn(0x300, 0x310);
+
+        transactions.addAll(Arrays.asList(
+                null,
+                null,
+                null,
+                transaction3,
+                transaction2,
+                transaction3,
+                transaction1
+        ));
+        cutPurge.mutate(inputCells, transactions, outputCells);
+        assertThat(outputCells, hasSize(3));
+        assertThat(outputCells, equalTo(Arrays.asList(
+                SITestUtils.getMockCommitCell(0x300),
+                SITestUtils.getMockAntiTombstoneCell(0x300),
+                SITestUtils.getMockValueCell(0x300))));
+    }
+
+    @Test
+    public void committed() {
+        TxnView committedWithRootParent = TxnTestUtils.getMockCommittedTxn(0x100, 0x200);
+        assertTrue(SICompactionStateMutate.committed(committedWithRootParent));
```

Comment:
Can you rename `SICompactionStateMutate.committed` to `isCommitted` ? I think that would be clearer.

Reply:
done